### ISSUE TITLE
Provide a default implementation for `IHostEnvironmentStatistics` for Windows & Linux

### ISSUE THREAD

**Issue by ReubenBond** [dotnet/orleans#8270]
`IHostEnvironmentStatistics` provides CPU and memory statistics to Orleans, which it used to trigger load shedding.
Currently, we do not enable any implementations of this by default and instead log a warning during startup. This is not a good experience.

We should add a Windows-compatible implementation and configure an appropriate implementation by default, depending on the OS (eg, `RuntimeInformation.IsOSPlatform(OSPlatform.Linux)`.

**Comment by AlgorithmsAreCool** [comment]
As a note to anyone concerned. I think this can be implemented in a platform agnostic way in .NET Core 6+
The new-ish ` GC.GetGCMemoryInfo()` API returns some great data from the gc/runtime about memory size and pressure.

For CPU usage, I tried using the new Metrics/MeterListener API, but it seems the System.Runtime counters are still published via the older EventCounter API, which is gross and poorly documented. But I have a rough sketch working.

I am not 100% sure I'm filtering the EventListener correctly to minimize overhead, I'll toy with it some more later.

```csharp
public class HostEnvironmentStatistics : EventListener, IHostEnvironmentStatistics
{
    public HostEnvironmentStatistics(ILogger<HostEnvironmentStatistics> logger)
    {
        Logger = logger;
    }

    private ILogger<HostEnvironmentStatistics> Logger { get; }

    public long? TotalPhysicalMemory => GC.GetGCMemoryInfo().TotalAvailableMemoryBytes;
    public float? CpuUsage { get; private set; }
    public long? AvailableMemory
    {
        get
        {
            var gcInfo = GC.GetGCMemoryInfo();
            return gcInfo.TotalAvailableMemoryBytes - gcInfo.MemoryLoadBytes;
        }
    }

    //BEWARE, THIS METHOD IS CALLED BEFORE THE CTOR!!
    protected override void OnEventSourceCreated(EventSource eventSource)
    {
        if (eventSource.Name == "System.Runtime")
        {
            var argDictionary = new Dictionary<string, string?>()
            {
                ["EventCounterIntervalSec"] = "1"
            };

            EnableEvents(eventSource, EventLevel.Informational, EventKeywords.None, argDictionary);
        }
    }

    //BEWARE, THIS METHOD IS CALLED BEFORE THE CTOR!!
    protected override void OnEventWritten(EventWrittenEventArgs eventData)
    {
        if (Logger is not null)
        {
            if(eventData is {  EventName: "EventCounters", Payload: IReadOnlyList<object?> payload })
            {

                var eventPayload = payload.OfType<IDictionary<string, object>>().FirstOrDefault();

                if(
                    eventPayload is not null &&
                    eventPayload.TryGetValue("Name", out var counterNameObject) &&
                    counterNameObject is string counterName &&
                    counterName == "cpu-usage"
                    )
                {
                    if (eventPayload.TryGetValue("Mean", out object? meanValueObject) && meanValueObject is double meanValue)
                    {
                        CpuUsage = (float)meanValue;
                    }
                }
            }
        }
    }
}
```

**Comment by ledjon-behluli** [comment]
@ReubenBond i think this can be closed now

**Comment by ReubenBond** [comment]
Yes, thank you!
Resolved by #8820
